src/main/java/net/minecraftforge/common/BiomeDictionary.java

```diff
@@ -249,251 +219,238 @@ public static boolean areBiomesEquivalent(Biome biomeA, Biome biomeB)
     }
 
     /**
-     * Checks to see if the given biome is registered as being a specific type
+     * Checks to see if the given biome is has been tagged with the given type.
      *
-     * @param biome the biome to be considered
-     * @param type the type to check for
-     * @return returns true if the biome is registered as being of type type, false otherwise
+     * @param biome the biome
+     * @param type  the type to check for
+     * @return true if the biome has been tagged with the type, false otherwise
      */
-    public static boolean isBiomeOfType(Biome biome, Type type)
+    public static boolean hasType(Biome biome, Type type)
     {
-        checkRegistration(biome);
-        return containsType(getBiomeInfo(biome), type);
+        return getTypes(biome).contains(type);
     }
 
     /**
-     * Checks to see if the given biome has been registered as being of any type
+     * Checks to see if the given biome has been tagged with any type.
+     *
      * @param biome the biome to consider
-     * @return returns true if the biome has been registered, false otherwise
+     * @return true if the biome has been tagged with any type
      */
-    public static boolean isBiomeRegistered(Biome biome)
+    public static boolean hasTypes(Biome biome)
     {
-        return biomeInfoMap.containsKey(Biome.REGISTRY.getNameForObject(biome));
+        return !getBiomeInfo(biome).types.isEmpty();
     }
 
-    public static void registerAllBiomes()
-    {
-        FMLLog.warning("Redundant call to BiomeDictionary.registerAllBiomes ignored");
-    }
     /**
-     * Loops through the biome list and automatically adds tags to any biome that does not have any
-     * This is called by Forge at postinit time. It will additionally dispatch any deferred decorator
-     * creation events.
+     * Automatically tags a given biome with appropriate types based on certain heuristics.
+     * If a biome's types are requested and the biome has not yet been tagged with any types, the biome's tags will be
```

Comment:
If a biome's types are requested and the biome has not had any types added, the biome's types will be